### PR TITLE
Check if valid access key/secret key before calling

### PR DESCRIPTION
**Pull request by supriti**
s3connect

Signed-off-by: Supriti Singh <supriti.singh@suse.com>

### CODE REVIEW

File: srv/salt/_modules/rgw.py [hunk 1]
```diff
@@ -230,6 +230,11 @@ def endpoints(cluster='ceph'):
 
 
 def s3connect(user):
+    """
+    Return an S3 connection
+    """
+    if access_key(user) is None or secret_key(user) is None:
```
Comment (jschmid1) [inline srv/salt/_modules/rgw.py]: more pythonic:

``` python
if not smth:
  do_smth
```
You can omit the `is None` unless you specifically expect `None`. `if not` looks for `False`, `None` and empty strings.

File: srv/salt/_modules/rgw.py [hunk 2]
```diff
@@ -230,6 +230,11 @@ def endpoints(cluster='ceph'):
...
+        return
     endpoint = endpoints()[0]
 
     s3conn = boto.connect_s3(
```
Comment (jschmid1) [inline srv/salt/_modules/rgw.py]: @theanalyst That's the boto(3) call we have to adapt, right?

File: srv/salt/_modules/rgw.py (hunk 1 above)
Comment (supriti) [reply srv/salt/_modules/rgw.py]: I am specifically looking for None here. This function (https://github.com/SUSE/DeepSea/blob/master/srv/salt/_modules/rgw.py#L146) can return None and empty strings. Empty strings are ok, but None is not.

File: srv/salt/_modules/rgw.py (hunk 2 above)
Comment (supriti) [reply srv/salt/_modules/rgw.py]: @jschmid1connect_s3 complaints only when None is passed for access_key/secret_key(https://github.com/SUSE/DeepSea/issues/910). So exit early in that case.

Comment (theanalyst) [reply srv/salt/_modules/rgw.py]: yes, for newer versions we can move this to boto3

File: srv/salt/_modules/rgw.py (hunk 1 above)
Comment (jschmid1) [reply srv/salt/_modules/rgw.py]: ok, thanks. Haven't looked that far. No objections then.

**Comment by supriti** [comment]
@theanalyst @jschmid1 @swiftgist can this pr be merged?

**Comment by jschmid1** [comment]
please run rgw suite

**Comment by jschmid1** [comment]
please run ganesha suite

**Comment by jschmid1** [comment]
@smithfarm @kshtsk 

both testsuites report success although they timed out.

**Comment by smithfarm** [comment]
@jschmid1 Can you post a link to the log, so I can look at the timeout?

**Comment by jschmid1** [comment]
@smithfarm 
http://storage-ci.suse.de:8080/job/deepsea-teuthology-ganesha-trigger/6/default/artifact/teuth-deepsea-b6.log
http://storage-ci.suse.de:8080/job/deepsea-teuthology-rgw-trigger/12/default/artifact/teuth-deepsea-b12.log

**Comment by kshtsk** [comment]
@jschmid1 
>> both testsuites report success although they timed out.
We need to add a check to the jenkins job, that if it didn't find any jobs, i.e. PASS+FAIL+DEAD=0 it should return false. Or handle an exit code.

**Comment by kshtsk** [comment]
Pulpito at http://158.69.67.200:8081/ is down, this is the reason of this failure.

**Comment by kshtsk** [comment]
please run rgw suite

**Comment by kshtsk** [comment]
please run ganesha suite

**Pull request merged**